src/Microsoft.Crank.Jobs.K6/Program.cs

```diff
@@ -0,0 +1,367 @@
+            {
+                try
+                {
+                    process.Start();
+                    break;
+                }
+                catch (Exception e)
+                {
+                    // The process might fail with a permissions exception during unit tests
+                    await Task.Delay(500);
+                    Console.WriteLine("Error, retrying: " + e.Message);
+                }
+            }
+        }
+
+        private static bool TryGetArgumentValue<T>(string argName, List<string> argsList, out T value)
+        {
+            var argumentIndex = argsList.FindIndex(arg => string.Equals(arg, argName, StringComparison.OrdinalIgnoreCase));
+            if (argumentIndex >= 0)
+            {
+                string copy = argsList[argumentIndex + 1];
+                argsList.RemoveAt(argumentIndex);
+                argsList.RemoveAt(argumentIndex);
+
+                try
+                {
+                    value = (T) Convert.ChangeType(copy, typeof(T));
+                    return true;
+                }
+                catch
+                {
+                }
+            }
+
+            value = default(T);
+            return false;
+        }
+
+        public static async Task MeasureFirstRequest(string[] args)
+        {
+            var url = args.FirstOrDefault(arg => arg.StartsWith("URL=", StringComparison.OrdinalIgnoreCase));
+
+            if (url == null)
+            {
+                Console.WriteLine("URL not found, skipping first request");
+                return;
+            }
+
+            url = url.Substring(4);
+
+            var cts = new CancellationTokenSource(30000);
+            var httpMessage = new HttpRequestMessage(HttpMethod.Get, url);
+
+            var stopwatch = Stopwatch.StartNew();
+
+            try
+            {
+                using (var response = await _httpClient.SendAsync(httpMessage, cts.Token))
+                {
+                    var elapsed = stopwatch.ElapsedMilliseconds;
+                    Console.WriteLine($"{elapsed} ms");
+
+                    BenchmarksEventSource.Register("http/firstrequest", Operations.Max, Operations.Max, "First Request (ms)", "Time to first request in ms", "n0");
+                    BenchmarksEventSource.Measure("http/firstrequest", elapsed);
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                Console.WriteLine("A timeout occurred while measuring the first request");
+            }
+            catch (HttpRequestException)
+            {
+                Console.WriteLine("A connection exception occurred while measuring the first request");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("An unexpected exception occurred while measuring the first request:");
+                Console.WriteLine(e.ToString());
+            }
+        }
+
+        private static string Quote(string s)
```

Comment:
most of the methods are in other projects as well. consider consolidating them in util project or something.